pkg/scheduler/plugins/coscheduling/core/gang_cache.go

```diff
@@ -144,16 +141,21 @@ func (gangCache *GangCache) onPodGroupAdd(obj interface{}) {
 	gangName := pg.Name
 
 	gangId := util.GetId(gangNamespace, gangName)
-	gang := gangCache.getGangFromCacheByGangId(gangId, false)
-	if gang == nil {
-		gang = gangCache.getGangFromCacheByGangId(gangId, true)
-		klog.Infof("Create gang by podGroup on add, gangName: %v", gangId)
-	}
-
+	gang := gangCache.getGangFromCacheByGangId(gangId, true)
 	gang.tryInitByPodGroup(pg, gangCache.pluginArgs)
 }
 
 func (gangCache *GangCache) onPodGroupUpdate(oldObj interface{}, newObj interface{}) {
+	pg, ok := newObj.(*v1alpha1.PodGroup)
+	if !ok {
+		return
+	}
+	gangNamespace := pg.Namespace
+	gangName := pg.Name
+
+	gangId := util.GetId(gangNamespace, gangName)
+	gang := gangCache.getGangFromCacheByGangId(gangId, true)
```

Comment:
Gang object must be exist when got Update Event, so here call `getGangFromCacheByGangId` should pass the `false`.